src/m1_facilitator/Manifest/Manifest.ts

```diff
@@ -0,0 +1,259 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+import { Validator as JsonSchemaVerifier } from 'jsonschema';
+import schema from './manifest.schema.json';
+
+interface ManifestInputChain {
```

Comment:
Please add documentation and it can be better renamed e.g. ManifestChain ?

Reply:
Updated the naming and added verbose documentation.